From a8e31e3539298ad98bef8a2d03f91cbbaeee73c2 Mon Sep 17 00:00:00 2001
Message-Id: <RFC-patch-1.2-a8e31e35392-20211117T112332Z-avarab@gmail.com>
In-Reply-To: <RFC-cover-0.2-00000000000-20211117T112332Z-avarab@gmail.com>
References: <20211117005701.371808-1-emilyshaffer@google.com>
	<RFC-cover-0.2-00000000000-20211117T112332Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Wed, 17 Nov 2021 12:18:53 +0100
Subject: [RFC PATCH 1/2] submodule tests: fix potentially broken "config ..
 --unset"
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

These "config ... --unset" at the start must be guarded by something
like a test_might_fail, or we'll fail if a previous test didn't run,
e.g. due to the --run option.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 t/t7406-submodule-update.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/t/t7406-submodule-update.sh b/t/t7406-submodule-update.sh
index b42a339982b..01e1acaf300 100755
--- a/t/t7406-submodule-update.sh
+++ b/t/t7406-submodule-update.sh
@@ -1063,7 +1063,7 @@ test_expect_success 'submodule update --quiet passes quietness to fetch with a s
 
 test_expect_success 'submodule update adds superproject gitdir to older repos' '
 	(cd super &&
-	 git -C submodule config --unset submodule.superprojectGitdir &&
+	 test_might_fail git -C submodule config --unset submodule.superprojectGitdir &&
 	 git submodule update &&
 	 test-tool path-utils relative_path \
 		"$(git rev-parse --absolute-git-dir)" \
@@ -1075,7 +1075,7 @@ test_expect_success 'submodule update adds superproject gitdir to older repos' '
 
 test_expect_success 'submodule update uses config.worktree if applicable' '
 	(cd super &&
-	 git -C submodule config --unset submodule.superprojectGitDir &&
+	 test_might_fail git -C submodule config --unset submodule.superprojectGitDir &&
 	 git -C submodule config extensions.worktreeConfig true &&
 	 git submodule update &&
 	 test-tool path-utils relative_path \
-- 
2.34.0.796.g2c87ed6146a

